### ISSUE TITLE
[$2000] iOS - Chat - User is directed to the last accessed conversation instead of Concierge chat

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#19138]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Access a conversation from the chat
2. Log out of NewDot
3. Click on a link that is sent in another app: https://staging.new.expensify.com/concierge
4. Verify that you are taken to the login screen
5. Log in to NewDot
6. Verify you're navigated to the concierge chat after login

## Expected Result:
User should be directed to the Concierge chat after login since they clicked on concierge conv link before login

## Actual Result:
User is directed to the last accessed conversation before logout, instead of concierge chat

## Workaround:
unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [X] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.15.0
**Reproducible in staging?:** Y
**Reproducible in production?:** N (links are not working in IOS) 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
![Bug6058914_Sc1_ND](https://github.com/Expensify/App/assets/43995119/59caba62-74cb-479e-b412-287344e124ac)

https://github.com/Expensify/App/assets/43995119/5fe19164-e26e-4721-afbd-5eecf5219ede

**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team 
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e55e86b6d93829ac</li>
        <li>Upwork Job ID: 1664037836915187712</li>
        <li>Last Price Increase: 2023-06-06</li>
    </ul>
</details>

**Comment by twisterdotcom** [comment]
>Verify you're navigated to the concierge chat after login

This seems like it's taken from a TestRail case, am I correct @mvtglobally? If so, I wonder why we need this to be the case.

**Comment by twisterdotcom** [comment]
Slack: https://expensify.slack.com/archives/C03U7DCU4/p1684426512880779

**Comment by twisterdotcom** [comment]
Okay, @mvtglobally - could we amend whatever testrail case this is? Landing in your last chat is the expected way, so going to assign this to you so you can update testrail.

**Comment by mvtglobally** [comment]
@twisterdotcom can you help me with what exact steps/ actions are no longer relevant?

**Comment by twisterdotcom** [comment]
Okay, damn I lost this because I unassigned. Taking it back. @neil-marcellini set me straight here that because you clicked from `/concierge`, you should land in `/concierge`, even if it's after signing in.

**Comment by jasperhuangg** [comment]
Hmmm I've worked on this logic before and happy to take over on this one and handle it internally

**Comment by jasperhuangg** [comment]
For anyone looking into this, the logic for this lives [here](https://github.com/Expensify/App/blob/main/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js#L65-L81)

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel I'm gonna be going OOO next week, so if you start receiving proposals and I'm not around to help review them feel free to ask another internal engineer to help you move things forward.

**Comment by twisterdotcom** [comment]
Waiting on proposals.

**Comment by twisterdotcom** [comment]
Bumping the amount

**Comment by cubuspl42** [comment]
I wasn't able to reproduce

**Comment by twisterdotcom** [comment]
Okay @cubuspl42 - I agree, now I can't.

I do not know what fixed this, but I'm going to close.

https://github.com/Expensify/App/assets/9133401/afb7d6ed-0083-46ee-adac-3ec51af6ae71
